MCR-2398 MCRXMLMetadataManager Interface

OPNA2608 opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-2398).

~~Still need to test & find a good solution for code calling `MCRXMLMetadataManager.instance()` directly vs `MCRAbstractXMLMetadataManager.getManager()` reading the configured XML Metadata Manager class and instantiating it via its `instance()` method.~~

Fixed by moving MCRXMLMetadataManager -> MCRDefaultXMLMetadataManager & MCRAbstractXMLMetadataManager -> MCRXMLMetadataManager.

MyCoRe-Travis commented:
Something gone wrong on eb60ce75433cd952ac559146840f3e406434b150! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7087-pull_request-1234

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

OPNA2608 commented:
Gonna look over code comments again, otherwise ready for review.

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Improve MCRXMLMetadataManager documentation, make it specific to the abstract new class.

MyCoRe-Travis commented:
Something gone wrong on 1c68194d23490cea10dc3477f1e83b96b63841fc! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7122-pull_request-1234

rsteph-de commented on mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java:
```diff
@@ -47,44 +32,44 @@
- * MCR.Metadata.Store.DefaultClass=org.mycore.datamodel.ifs2.MCRMetadataStore
+ * 
+ * If an SVN-based store is configured, then the following property will be used to
+ * store & manage local SVN repositories:
```
error: bad HTML entity:
* store & manage local SVN repositories:

OPNA2608 replied:
Fixed, thanks.

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Fixed malformed documentation.

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Improve javadocs.

MyCoRe-Travis commented:
Something gone wrong on 5a8bd8504a8e669b7e56322b7f1b977cf64677ef! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7140-pull_request-1234

MyCoRe-Travis commented:
Something gone wrong on fd38519db19d3a9dc595fce5d248e858e3ae0e68! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7144-pull_request-1234

MyCoRe-Travis commented:
Something gone wrong on d0d8462ec22545b23fe39fa074bc7ddfb18d3d7f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7146-pull_request-1234

yagee-de reviewed (approved):
Looks good as the next step for me. But I think one refactoring would be useful:

- move all abstract methods to an interface and let MCRDefaultXMLMetadataManager inherit this interface and remove the superclass from MCRDefaultXMLMetadataManager
- make abstract methods public and delegate to the interface methods

Interfaces are a better preparation for java modules

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Convert MCRXMLMetadataManager to an interface as requested in PR comments.

OPNA2608 commented:
I've dropped all commits that were unrelated to the ticket's goal. I'll move them into a follow-up PR once this is merged.

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Fix misunderstanding about desired class structure.

sebhofmann commented on mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java:
```diff
@@ -18,379 +18,289 @@
+            SINGLETON = new MCRXMLMetadataManager();
+        }
+        if (IMPLEMENTATION == null) {
+            IMPLEMENTATION = MCRConfiguration2
```
The Property MCR.Metadata.Manager.Class should be defined in the mycore.properties.
We have the Rule that every default property should be in the mycore.properties, except for dynamic named properties.

sebhofmann reviewed (changes requested):
Everything else looks good for me.

OPNA2608 committed:
MCR-2398 MCRXMLMetadataManager Interface

Define new MCR.Metadata.Manager.Class property in mycore.properties.
